File: src/clean.py
#!/usr/bin/env python3
"""
Script to remove build artifacts

Delete build/ and dist/ directories
"""

import shutil
import sys
from pathlib import Path

from src.logger import logger
from src.paths import BUILD_DIR, DIST_DIR


def clean_directory(path: Path) -> None:
    """
    Delete a directory

    Args:
        path: Path of the directory to delete
    """
    if path.exists():
        logger.info(f"Removing {path}/")
        try:
            shutil.rmtree(path)
            logger.info(f"Removed {path}/")
        except Exception as e:
            logger.error(f"Failed to remove {path}/: {e}")
            sys.exit(1)
    else:
        logger.info(f"{path}/ does not exist (skipped)")


def main():
    logger.info("Cleaning build artifacts")

    for directory in (BUILD_DIR, DIST_DIR):
        clean_directory(directory)

    logger.info("Clean complete")


if __name__ == "__main__":
    main()

File: crates/cli/src/clean.rs
use anyhow::Result;
use std::fs;
use std::path::Path;

pub fn clean(build_dir: &Path, dist_dir: &Path) -> Result<()> {
    let mut removed = 0;

    if build_dir.exists() {
        fs::remove_dir_all(build_dir)?;
        println!("Removed {}", build_dir.display());
        removed += 1;
    } else {
        println!("Skipped {} (not found)", build_dir.display());
    }

    if dist_dir.exists() {
        fs::remove_dir_all(dist_dir)?;
        println!("Removed {}", dist_dir.display());
        removed += 1;
    } else {
        println!("Skipped {} (not found)", dist_dir.display());
    }

    println!("Cleaned {removed} directories");
    Ok(())
}
